cs/src/Connections/ConnectionStatus.cs

```diff
@@ -34,4 +34,9 @@ public enum ConnectionStatus
     /// Disconnected from the service and could not reconnect either due to disposal, service down, tunnel deleted, or token expiration. This is the final status.
     /// </summary>
     Disconnected,
+
+    /// <summary>
+    /// Refreshing tunnel host public key.
+    /// </summary>
+    RefreshingTunnelHostPublicKey,
```

Comment:
Do we need a separate status value for this? Should it just be part of `Connecting` status?

Reply:
Since there is `RefreshingTunnelAccessToken` which is also part of (re-)connection flow, I though adding `RefreshingTunnelHostPublicKey` may help with transparency into the current client status, and with unit tests.

Reply:
> there is `RefreshingTunnelAccessToken`

I forgot we had that one already. I'm a little concerned that as we're likely to add/remove/reorder different (optional) sub-steps of connecting, maybe that detail shouldn't be exposed to the application. But maybe it's fine.